Sources/XCTest/Public/XCTAssert.swift

```diff
@@ -421,3 +423,14 @@ public func XCTAssertThrowsError<T>(_ expression: @autoclosure () throws -> T, _
         }
     }
 }
+
+public func XCTAssertNoThrow<T>(_ expression: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    _XCTEvaluateAssertion(.noThrow, message: message, file: file, line: line) {
+        do {
+             _ = try expression()
+            return .success
+        } catch _ {
+            return .expectedFailure("threw error")
```

Comment:
@briancroom Re; textual description, I was just following what was done for `XCTAsserThrowsError`: https://github.com/apple/swift-corelibs-xctest/blob/240d0bccd4eb39aab1c900fedaf79dc4ce724820/Sources/XCTest/Public/XCTAssert.swift#L420

I'm more than happy to make it more descriptive as I did in the overlay you referenced.